Content.Trauma.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,690 @@
+    {
+        if (TryComp<SeedComponent>(target, out var seedComp))
+        {
+            if (seedComp.Seed != null)
+            {
+                seedComp.Seed.Mutations.Clear();
+            }
+            else
+            {
+                if (seedComp.Seed == null || seedComp.SeedId == null)
+                    return;
+                _prototypeManager.TryIndex(seedComp.SeedId, out SeedPrototype? protoSeed);
+                if (protoSeed == null)
+                    return;
+                seedComp.Seed = protoSeed.Clone();
+                seedComp.Seed.Mutations.Clear();
+            }
+        }
+        else if (TryComp<PlantHolderComponent>(target, out var plantComp))
+        {
+            if (plantComp.Seed != null)
+            {
+                plantComp.Seed.Mutations.Clear();
+            }
+        }
+    }
+    public void ReadScannedPlant(Entity<PlantAnalyzerComponent> ent, EntityUid target)  //Funkystation - Renamed to match plants instead of copying HealthAnalyzer func names
+    {
+
+        if (TryComp<SeedComponent>(target, out var seedComp))
+        {
+            if (seedComp.Seed != null)
+            {
+                var state = ObtainingGeneDataSeed(seedComp.Seed, target, false);
+                _uiSystem.SetUiState(ent.Owner, PlantAnalyzerUiKey.Key, state);  //Funkystation - Swapped to set state instead of UI message
+            }
+            else if (seedComp.SeedId != null && _prototypeManager.TryIndex(seedComp.SeedId, out SeedPrototype? protoSeed))
+            {
+                var state = ObtainingGeneDataSeed(protoSeed, target, false);
+                _uiSystem.SetUiState(ent.Owner, PlantAnalyzerUiKey.Key, state); //Funkystation - Swapped to set state instead of UI message
+            }
+        }
+        else if (TryComp<PlantHolderComponent>(target, out var plantComp))
+        {
+            if (plantComp.Seed != null)
+            {
+                var state = ObtainingGeneDataSeed(plantComp.Seed, target, true);
+                _uiSystem.SetUiState(ent.Owner, PlantAnalyzerUiKey.Key, state); //Funkystation - Swapped to set state instead of UI message
+            }
+        }
+    }
+
+    /// <summary>
+    ///     Analysis of seed from prototype.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeed(SeedData seedData, EntityUid target, bool isTray)
+    {
+        // Get trickier fields first.
+        AnalyzerHarvestType harvestType = (AnalyzerHarvestType) seedData.HarvestRepeat;
+
+        var mutationProtos = seedData.MutationPrototypes;
+        List<string> mutationStrings = new();
+        foreach (var mutationProto in mutationProtos)
+        {
+            if (_prototypeManager.TryIndex<SeedPrototype>(mutationProto, out var seed))
+            {
+                mutationStrings.Add(seed.DisplayName);
+            }
+        }
+
+        PlantAnalyzerScannedSeedPlantInformation ret = new()
```

Comment:
just return it instead of assigning to intermediate variable that isnt used for anything